sample-code/src/orchestration.ts

```diff
@@ -1,114 +1,154 @@
 
+const llm: LlmModuleConfig = {
+  model_name: 'gpt-4o',
+  model_params: {}
+};
+
 /**
  * Ask about the capital of any country using a template.
- * @returns The message content from the orchestration service in the generative AI hub.
+ * @returns The orchestration service response.
  */
-async function orchestrationCompletionTemplate(): Promise<string | undefined> {
+export async function orchestrationTemplating(): Promise<OrchestrationResponse> {
   const orchestrationClient = new OrchestrationClient({
-    llm: {
-      model_name: 'gpt-4-32k',
-      model_params: {}
-    },
+    llm,
     templating: {
       template: [
+        // define "country" as variable by wrapping it with "{{? ... }}"
         { role: 'user', content: 'What is the capital of {{?country}}?' }
       ]
     }
   });
 
-  // Call the orchestration service.
-  const response = await orchestrationClient.chatCompletion({
+  return orchestrationClient.chatCompletion({
+    // give the actual value for the variable "country"
     inputParams: { country: 'France' }
   });
-  // Access the response content.
-  return response.getContent();
 }
 
+const templating = { template: [{ role: 'user', content: '{{?input}}' }] };
+
 /**
- * Allow any user input and apply input and output filters.
- * Handles the case where the input or output are filtered:
- * - In case the input was filtered the response has a non 200 status code.
- * - In case the output was filtered `response.getContent()` throws an error.
- * @returns The message content from the orchestration service in the generative AI hub.
+ * Apply a content filter to LLM requests, filtering any hateful input.
  */
-async function orchestrationCompletionFiltering(): Promise<string | undefined> {
+export async function orchestrationInputFiltering(): Promise<void> {
+  // create a filter with minimal thresholds for hate and violence
+  // lower numbers mean more strict filtering
   const filter = buildAzureContentFilter({ Hate: 0, Violence: 0 });
   const orchestrationClient = new OrchestrationClient({
-    llm: {
-      model_name: 'gpt-4-32k',
-      model_params: {}
-    },
-    templating: {
-      template: [{ role: 'user', content: '{{?input}}' }]
-    },
+    llm,
+    templating,
+    // configure the filter to be applied for both input and output
     filtering: {
-      input: filter,
-      output: filter
+      input: filter
     }
   });
 
   try {
-    // Call the orchestration service.
-    const response = await orchestrationClient.chatCompletion({
+    // trigger the input filter, producing a 400 - Bad Request response
+    await orchestrationClient.chatCompletion({
       inputParams: { input: 'I hate you!' }
     });
-    // Access the response content.
-    return response.getContent();
+    throw new Error('Input was not filtered as expected.');
   } catch (error: any) {
-    // Handle the case where the output was filtered.
-    return `Error: ${error.message}`;
+    if (error.response.status === 400) {
```

Comment:
[pp] Also, typecheck if `error instanceof Error`. Ideally, we should also check if it is an `AxiosError` as we are assuming it's structure. But, I guess this would lead to adding a new dependency.